recipes/liburing/all/conanfile.py

```diff
@@ -60,49 +56,57 @@ def validate(self):
-        if self.options.get_safe("with_libc") == False:
-            args.append("--nolibc")
-        self._autotools.configure(args=args)
-        self._autotools.flags.append("-std=gnu99")
-        return self._autotools
-
-    def _patch_sources(self):
-        for data in self.conan_data.get("patches", {}).get(self.version, []):
-            patch(self, **data)
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = AutotoolsToolchain(self)
+        def whether(condition):
+            return "" if condition else None
+        tc.update_configure_args({
+            "--nolibc": whether(not self.options.get_safe("with_libc", default=True)),
+            "--enable-shared": None,
+            "--disable-shared": None,
+            "--enable-static": None,
+            "--disable-static": None,
+            "--bindir": None,
+            "--sbindir": None,
+            "--oldincludedir": None,
+        })
+        tc.generate()
 
     def build(self):
-        self._patch_sources()
-        with chdir(self, self._source_subfolder):
-            autotools = self._configure_autotools()
-            autotools.make()
+        apply_conandata_patches(self)
+        with chdir(self, self.source_folder):
+            at = Autotools(self)
+            at.configure()
+            at.make(target="src")
 
     def package(self):
-        copy(self, "COPYING*", src=self._source_subfolder, dst=os.path.join(self.package_folder, "licenses"))
-        copy(self, "LICENSE", src=self._source_subfolder, dst=os.path.join(self.package_folder, "licenses"))
+        copy(self, "COPYING*", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
 
-        with chdir(self, self._source_subfolder):
-            autotools = self._configure_autotools()
-            install_args = [
-                "ENABLE_SHARED={}".format(1 if self.options.shared else 0)
-            ]
-            autotools.install(args=install_args)
+        with chdir(self, self.source_folder):
+            at = Autotools(self)
+            at.install(
+                args=[f"ENABLE_SHARED={1 if self.options.shared else 0}"]
+            )
 
         rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
         rmdir(self, os.path.join(self.package_folder, "man"))
 
         if self.options.shared:
             os.remove(os.path.join(self.package_folder, "lib", "liburing.a"))
+        
+            # FIXME: The liburing.so symlink ends up broken. Remove and replace.
+            symlinks.remove_broken_symlinks(self, self.package_folder)
+            source_file_suffix = "1" if self.version < Version("2.0") else "2"
+            os.symlink(src=f"liburing.so.{source_file_suffix}",
+                       dst=os.path.join(self.package_folder, "lib", "liburing.so"))
```

Comment:
I think that this could be solved by adding this line in the configure step (`tc.update_configure_args`):

```python
     "--libdevdir": "${prefix}/lib"  # pointing to libdir
```
Let's try that. If it works, we can remove all these lines.

Reply:
That seems to work, thanks!